eventsourcing/src/test/java/org/axonframework/eventsourcing/DefaultSnapshotterSpanFactoryTest.java

```diff
@@ -0,0 +1,83 @@
+package org.axonframework.eventsourcing;
+
+import org.axonframework.tracing.IntermediateSpanFactoryTest;
+import org.axonframework.tracing.SpanFactory;
+import org.axonframework.tracing.TestSpanFactory;
+import org.junit.jupiter.api.Test;
+
+
+class DefaultSnapshotterSpanFactoryTest extends IntermediateSpanFactoryTest<DefaultSnapshotterSpanFactory.Builder, DefaultSnapshotterSpanFactory> {
+
+    @Test
+    void testCreateScheduleSnapshotSpanWithDefaults() {
+        test(builder -> builder,
+                spanFactory -> spanFactory.createScheduleSnapshotSpan("MyAggregateType", "3728973982"),
+                new SpanDefinition("scheduleSnapshot(MyAggregateType)", TestSpanFactory.TestSpanType.INTERNAL)
+                        .addAttribute("aggregateIdentifier", "3728973982")
+        );
+    }
+
+    @Test
+    void testCreateScheduleSnapshotSpanIncludesAggregateName() {
+        test(builder -> builder.aggregateTypeInSpanName(true).separateTrace(false),
+                spanFactory -> spanFactory.createScheduleSnapshotSpan("MyAggregateType", "3728973982"),
+                new SpanDefinition("scheduleSnapshot(MyAggregateType)", TestSpanFactory.TestSpanType.INTERNAL)
+                        .addAttribute("aggregateIdentifier", "3728973982")
+        );
+    }
+
+    @Test
+    void testCreateScheduleSnapshotSpanDoesntIncludeAggregateName() {
+        test(builder -> builder.aggregateTypeInSpanName(false).separateTrace(false),
+                spanFactory -> spanFactory.createScheduleSnapshotSpan("MyAggregateType", "3728973982"),
+                new SpanDefinition("scheduleSnapshot", TestSpanFactory.TestSpanType.INTERNAL)
+                        .addAttribute("aggregateIdentifier", "3728973982")
+        );
+    }
+
+    @Test
+    void testCreateScheduleSnapshotSpanIsNotAffectedBySeparateTrace() {
+        test(builder -> builder.aggregateTypeInSpanName(false).separateTrace(true),
+                spanFactory -> spanFactory.createScheduleSnapshotSpan("MyAggregateType", "3728973982"),
+                new SpanDefinition("scheduleSnapshot", TestSpanFactory.TestSpanType.INTERNAL)
+                        .addAttribute("aggregateIdentifier", "3728973982")
+        );
+    }
+
+    @Test
+    void testCreateCreateTraceWithDefaults() {
```

Comment:
Nit: why don't the following tests refer the method they validate like the previous tests do?

Reply:
Problem between keyboard and chair